haproxy/haproxy_cmd/run.go

```diff
@@ -128,17 +128,20 @@ func CheckEnvironment(dataplaneapiBin, haproxyBin string) error {
 		currVer, e := getVersion(path)
 		if e != nil {
 			err = e
+			return
```

Comment:
* `return nil` ? (to explicitly match the contract, as it done at the end)
* do we need to "break" the flow here instead of doing what was done before?

Reply:
> * `return nil` ? (to explicitly match the contract, as it done at the end)
The `return nil` was for the parent function - here we want to exit the nest function.
Also, the nested function doesn't return anything, it just initializes the err variable from the parent function.

>     * do we need to "break" the flow here instead of doing what was done before?
Yes, because, we want to print the error in case of error other than the version comparison. For ex, when the binary doesn't exist.

Reply:
* true, my bad.
* Ack!